supervisor/container.go

```diff
@@ -30,9 +30,16 @@ type Container struct {
 	ownerPod *HyperPod
 }
 
-func (c *Container) run(p *Process) {
+func (c *Container) run(p *Process) error {
+	// Receives the early result of the attempt to start the container,
+	// which ensures that we do not return before EventContainerStart
+	// is emitted.
```

Comment:
Yes, you are right, this part looks good to me, one tips, we can move c.start out of goroutine and leave c.wait inside goroutine to get rid of channel. 

bad news is vm.NewContainer in c.start doesn't guarantee container is started in vm, it just passes out the `new container` message, we still need another patch to fix this.